### ISSUE TITLE
[A11y]: Resize handles for addon panel and sidebar are not accessible

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#33979]
### Describe the bug

Quoting the spec:

> ## separator role
> A divider that separates and distinguishes sections of content or groups of menuitems.
> 
> There are two types of separators: a static [structure](https://www.w3.org/TR/wai-aria-1.2/#structure) that provides only a visible boundary and a focusable, interactive [widget](https://www.w3.org/TR/wai-aria-1.2/#widget) that is also moveable. If a separator is not focusable, it is revealed to [assistive technologies](https://www.w3.org/TR/wai-aria-1.2/#dfn-assistive-technology) as a static structural element. For example, a static separator can be used to help visually divide two groups of menu items in a menu or to provide a horizontal rule between two sections of a page.
> 
> Authors MAY make a separator focusable to create a [widget](https://www.w3.org/TR/wai-aria-1.2/#widget) that both provides a visible boundary between two sections of content and enables the user to change the relative size of the sections by changing the position of the separator. A variable separator widget can be moved continuously within a range, whereas a fixed separator widget supports only two discrete positions. Typically, a fixed separator widget is used to toggle one of the sections between expanded and collapsed states.
> 
> If the separator is focusable, authors MUST set the value of [aria-valuenow](https://www.w3.org/TR/wai-aria-1.2/#aria-valuenow) to a [number](https://www.w3.org/TR/wai-aria-1.2/#valuetype_number) reflecting the current position of the separator and update that value when it changes. Authors SHOULD also provide the value of [aria-valuemin](https://www.w3.org/TR/wai-aria-1.2/#aria-valuemin) if it is not 0 and the value of [aria-valuemax](https://www.w3.org/TR/wai-aria-1.2/#aria-valuemax) if it is not 100. If missing or not a number, the implicit values of these attributes are as follows:
> 
> * The implicit value of aria-valuemin is 0.
> * The implicit value of aria-valuemax is 100.
> * In applications where there is more than one focusable separator, authors SHOULD provide an accessible name for each one.
> 
> Elements with the role separator have an implicit [aria-orientation](https://www.w3.org/TR/wai-aria-1.2/#aria-orientation) value of horizontal.

For us, it means the
* drag handle must communicate an orientation (for sidebar, always horizontal; for addon panel, depends if panel is on the bottom or on the right)
* we must communicate min/max values (min always 0, max depends on space available in viewport)
* we must use aria-valuenow to communicate the current size as exposed in the layout submodule of the manager API
* we must provide aria labels so users know if they're resizing the sidebar or addon panel

Once the Drag component is improved, we must
* add stories to cover all relevant props
* change the addon panel and sidebar callee code to pass the new props

### Reproduction link

yarn && yarn task compile -s compile && yarn storybook:ui

### System

```bash
N/A
```
